.github/workflows/workflow.yml

```diff
@@ -24,26 +24,80 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@v2
 
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+
       - name: Use OCaml ${{ matrix.ocaml-version }}
         uses: avsm/setup-ocaml@v1
         with:
           ocaml-version: ${{ matrix.ocaml-version }}
 
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - name: Try to restore yarn cache
+        uses: actions/cache@v1
+        id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
         with:
-          node-version: ${{ matrix.node-version }}
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Get opam root directory path
+        id: opam-root-dir-path
+        run: echo "::set-output name=dir::$(opam config var root)"
+
+      - name: Try to restore .opam folder
+        id: cache-opam
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.opam-root-dir-path.outputs.dir }}
+          # opam cache is coupled to esy lock files, so both caches will be wiped whenever esy deps change
```

Comment:
not sure if there's a way to generate a "lock file" for opam and checking it in... for now it's tied to esy `index.json` file.

Reply:
opam lock or opam switch export